tests/codegen/vec-in-place.rs

```diff
@@ -90,3 +90,25 @@ pub fn vec_iterator_cast_deaggregate_fold(vec: Vec<Baz>) -> Vec<[u64; 4]> {
     // correct.
     vec.into_iter().map(|e| unsafe { std::mem::transmute(e) }).collect()
 }
+
+// CHECK-LABEL: @vec_iterator_cast_unwrap_drop
+#[no_mangle]
+pub fn vec_iterator_cast_unwrap_drop(vec: Vec<Wrapper<String>>) -> Vec<String> {
+    // CHECK-NOT: br i1 %{{.*}}, label %{{.*}}, label %{{.*}}
+    // CHECK-NOT: call void @__rust_dealloc
```

Comment:
to check that the IR is O(1) you should check the absence of calls in general, otherwise loops and branches could be hidden in a callee. It's unlikely to happen considering all the `#[inline]` we slap on everything, but I think it still expresses the intent better.

Reply:
Is there a way to allow random `@llvm.assume` or `@llvm.experimental.noalias.scope.decl` calls? These currently all get cleaned up, however I agree that your scenario is more likely to happen in the event of a regression.

If not, I can change these to just be `CHECK-NOT: call`.

Reply:
Afaik the filecheck syntax does not support negative lookbehind in its regular expressions. And the other tests also have `CHECK-NOT: call` so it should be fine.